### PR TITLE
Feat/df string representation function

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
# Pull Request: Fix and Enhance DataFrame to CSV String Conversion Function

## Overview

This pull request addresses the following:
1. Fixes the `_to_str` function to correctly sort DataFrame columns and rows.
2. Enhances the function to handle various data types, including dates, floats, and NaN values.
3. Adds comprehensive test cases to ensure the function works as expected across different scenarios.

### Function Update
- **df_to_str Function**: 
  - Sorts columns alphabetically.
  - Sorts rows based on all columns, ensuring NaN values are placed at the end.
  - Converts the DataFrame to a CSV string.

### Tests Added
1. **test_simple_multi_row**: Tests the function with a simple multi-row DataFrame.
2. **test_date_columns**: Tests the function with a DataFrame that includes date columns.
3. **test_float_columns**: Tests the function with a DataFrame that includes float columns.
4. **test_nan_values**: Tests the function with a DataFrame that includes NaN values using `pd.NA`.

## Conclusion

These changes improve the robustness and reliability of the `df_to_str` function, ensuring it works correctly with various data types and edge cases. The comprehensive tests provide confidence in the function's correctness and stability.

@lasuk Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

File: cashctrl_ledger/ledger_utils.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""
+This module provides utilities for handling and manipulating pandas DataFrames.
+"""
+
+import pandas as pd
+
+def df_to_str(target: pd.DataFrame) -> str:
+    """
+    Converts a DataFrame to a string, sorting columns and rows
+    for consistent comparison.
```
Comment (lasuk) [inline cashctrl_ledger/ledger_utils.py:10]: Let's insist on the fact that we provide a unique representation regardless of ordering.
Also let's state what this function is useful for.
(Good practice to pass by ChatGPT to beautify the text):
```
Converts a DataFrame to a unique string representation, regardless of the column or row order.
This function is helpful to identify exact or close matches between collections of data frames.
```

File: cashctrl_ledger/ledger_utils.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+"""
+This module provides utilities for handling and manipulating pandas DataFrames.
+"""
+
+import pandas as pd
+
+def df_to_str(target: pd.DataFrame) -> str:
```
Comment (lasuk) [inline cashctrl_ledger/ledger_utils.py:7]: How about adding an argument `index: bool =False`. If True, the index is included in the string representation.

This would make the function more generic and clearly document, that the index is ignored by default.

Comment (lasuk) [inline cashctrl_ledger/ledger_utils.py:7]: I am not convinced by the name `df_to_str`. It is too generic (`to_csv` is also converts a data frame to string) and does not convey the main point of the function, which is to translate the data frame to a unique, easily comparable form. Let's ask ChatGPT for better names.

**Review by lasuk — COMMENTED** [review]
Thanks for the function to convert a DataFrame into a unique string representation. 

Using `to_csv` is a pretty smart move: it is guaranteed to work with any data frame, is easily (human)interpretable while staying sufficiently concise.

I've added a few comments. Most importantly, the test suite does not cover the main point why we wrote this function. Let's revise it.

File: cashctrl_ledger/ledger_utils.py (hunk 2 above)
Comment (AlexTheWizardL) [reply cashctrl_ledger/ledger_utils.py:7]: Here is the suggested names I like the most:

`df_to_consistent_str`
`standardize_df_to_str`
`df_to_standardized_str`

Comment (AlexTheWizardL) [reply cashctrl_ledger/ledger_utils.py]: I`ve predicted this vote)

**Comment by lasuk** [comment]
I've pushed commit e49baa4 to improve test coverage.
Tests now ensure a data frame with same content but different row or column sequence leads to the same string representation.

@AlexTheWizardL, please shout if you disagree.

**Comment by lasuk** [comment]
Merged. Thanks a lot!
